cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSICartoDBSinkTest.java

```diff
@@ -48,208 +56,295 @@ public NGSICartoDBSinkTest() {
+     * [NGSICartoDBSink.configure] -------- Configured 'enable_distance' cannot be different than 'true' or 'false'.
      */
     @Test
-    public void testConfigureEnableLowercaseAlwaysTrue() {
+    public void testConfigureEnableDistanceOK() {
         System.out.println(getTestTraceHead("[NGSICartoDBSink.configure]")
-                + "-------- Independently of the configured value, enable_lowercase is always 'true' by default");
+                + "-------- Configured 'enable_distance' cannot be different than 'true' or 'false'");
         String endpoint = "https://localhost";
         String apiKey = "1234567890abcdef";
         String dataModel = null; // default one
-        String enableLowercase = "false";
+        String enableLowercase = null; // default one
         String flipCoordinates = null; // default one
         String enableRaw = null; // default one
-        String enableDistance = null; // default one
+        String keysConfFile = "/keys.conf";
+        String enableDistance = "falso";
         NGSICartoDBSink sink = new NGSICartoDBSink();
         sink.configure(createContext(endpoint, apiKey, dataModel, enableLowercase, flipCoordinates, enableRaw,
-                enableDistance));
+                enableDistance, keysConfFile));
         
         try {
-            assertTrue(sink.enableLowercase);
+            assertTrue(sink.invalidConfiguration);
             System.out.println(getTestTraceHead("[NGSICartoDBSink.configure]")
-                    + "-  OK  - 'enable_lowercase=false' was confiured, nevertheless it is always true by default");
+                    + "-  OK  - 'enable_distance=falso' was detected");
         } catch (AssertionError e) {
             System.out.println(getTestTraceHead("[NGSICartoDBSink.configure]")
-                    + "- FAIL - 'enable_lowercase=false' was confiured and it is really false");
+                    + "- FAIL - 'enable_distance=falso' was not detected");
             throw e;
         } // try catch
-    } // testConfigureEnableLowercaseAlwaysTrue
+    } // testConfigureEnableDistanceOK
     
     /**
-     * [NGSICartoDBSink.configure] -------- Configured `flip_coordinates` cannot be different than `true` or `false`.
+     * [NGSICartoDBSink.configure] -------- Configured 'keys_conf_file' cannot be empty.
      */
     @Test
-    public void testConfigureFlipCoordinatesOK() {
+    public void testConfigureKeysConfFileOK() {
         System.out.println(getTestTraceHead("[NGSICartoDBSink.configure]")
-                + "-------- Configured `flip_coordinates` cannot be different than `true` or `false`");
+                + "-------- Configured 'keys_conf_file' cannot be empty");
         String endpoint = "https://localhost";
         String apiKey = "1234567890abcdef";
         String dataModel = null; // default one
         String enableLowercase = null; // default one
-        String flipCoordinates = "falso";
+        String flipCoordinates = null; // default one
```

Comment:
Configuring something with `null` is the same than saying this parameter does not appear in the configuration. In this particular case, configuring `flip_coordinates` to null is the same than saying the configuration has no `flip_coordinates` parameter, thus, the `context.getString("flip_coordinates", "false")` gets the default value, which is `false`.